--- include/openssl/bio.h.orig	Mon Sep  5 22:30:32 2022
+++ include/openssl/bio.h	Tue Sep  6 01:38:22 2022
@@ -666,8 +666,24 @@ void BIO_copy_next_retry(BIO *b);
 
 /*long BIO_ghbn_ctrl(int cmd,int iarg,char *parg);*/
 
+#ifdef __MINGW_PRINTF_FORMAT
 int
 BIO_printf(BIO *bio, const char *format, ...)
+	__attribute__((__format__(__MINGW_PRINTF_FORMAT, 2, 3), __nonnull__(2)));
+int
+BIO_vprintf(BIO *bio, const char *format, va_list args)
+	__attribute__((__format__(__MINGW_PRINTF_FORMAT, 2, 0), __nonnull__(2)));
+int
+BIO_snprintf(char *buf, size_t n, const char *format, ...)
+	__attribute__((__deprecated__, __format__(__MINGW_PRINTF_FORMAT, 3, 4),
+	    __nonnull__(3)));
+int
+BIO_vsnprintf(char *buf, size_t n, const char *format, va_list args)
+	__attribute__((__deprecated__, __format__(__MINGW_PRINTF_FORMAT, 3, 0),
+	    __nonnull__(3)));
+#else
+int
+BIO_printf(BIO *bio, const char *format, ...)
 	__attribute__((__format__(__printf__, 2, 3), __nonnull__(2)));
 int
 BIO_vprintf(BIO *bio, const char *format, va_list args)
@@ -680,6 +696,7 @@ int
 BIO_vsnprintf(char *buf, size_t n, const char *format, va_list args)
 	__attribute__((__deprecated__, __format__(__printf__, 3, 0),
 	    __nonnull__(3)));
+#endif
 
 void ERR_load_BIO_strings(void);
 
