--- ssl/ssl_txt.c.orig	Wed Jun 22 11:37:59 2022
+++ ssl/ssl_txt.c	Wed Jun 22 11:37:59 2022
@@ -82,6 +82,7 @@
  * OTHERWISE.
  */
 
+#include <inttypes.h>
 #include <stdio.h>
 
 #include <openssl/buffer.h>
@@ -174,8 +174,8 @@
 	}
 
 	if (x->time != 0) {
-		if (BIO_printf(bp, "\n    Start Time: %lld",
-		    (long long)x->time) <= 0)
+		if (BIO_printf(bp, "\n    Start Time: %"PRId64,
+		    (int64_t)x->time) <= 0)
 			goto err;
 	}
 
